### ISSUE TITLE
Mount snap "gimp" (94) (cannot find required base "core18")

### ISSUE THREAD

**Issue by terotil** [snapcrafters/gimp#60]
Tried to install gimp using snap on Debian 9.6 (default kernel and desktop).

```
$ sudo apt install snapd 
[sudo] salasana henkilölle tero: 
Luetaan pakettiluetteloita... Valmis
Muodostetaan riippuvuussuhteiden puu       
Luetaan tilatiedot... Valmis        
The following packages were automatically installed and are no longer required:
  linux-image-4.9.0-4-amd64 linux-image-4.9.0-6-amd64
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  apparmor libapparmor-perl snap-confine squashfs-tools ubuntu-core-launcher
Ehdotetut paketit:
  apparmor-profiles apparmor-profiles-extra apparmor-utils
Seuraavat UUDET paketit asennetaan:
  apparmor libapparmor-perl snap-confine snapd squashfs-tools ubuntu-core-launcher
0 päivitetty, 6 uutta asennusta, 0 poistettavaa ja 0 päivittämätöntä.
Noudettavaa arkistoa 6 212 kt.
Toiminnon jälkeen käytetään 32,6 M t lisää levytilaa.
Haluatko jatkaa? [K/e] 
Nouda:1 http://ftp.fi.debian.org/debian stretch/main amd64 libapparmor-perl amd64 2.11.0-3+deb9u2 [82,2 kB]
Nouda:2 http://ftp.fi.debian.org/debian stretch/main amd64 apparmor amd64 2.11.0-3+deb9u2 [525 kB]
Nouda:3 http://ftp.fi.debian.org/debian stretch/main amd64 snap-confine amd64 2.21-2+b1 [65,2 kB]
Nouda:4 http://ftp.fi.debian.org/debian stretch/main amd64 ubuntu-core-launcher amd64 2.21-2+b1 [36,0 kB]
Nouda:5 http://ftp.fi.debian.org/debian stretch/main amd64 squashfs-tools amd64 1:4.3-3+deb9u1 [124 kB]
Nouda:6 http://gensho.ftp.acc.umu.se/debian stretch/main amd64 snapd amd64 2.21-2+b1 [5 380 kB]
Noudettiin 6 212 kt ajassa 1s (5 783 kt/s)
Esiräätälöidään paketteja...
Selecting previously unselected package libapparmor-perl.
(Reading database ... 199102 files and directories currently installed.)
Preparing to unpack .../0-libapparmor-perl_2.11.0-3+deb9u2_amd64.deb ...
Unpacking libapparmor-perl (2.11.0-3+deb9u2) ...
Selecting previously unselected package apparmor.
Preparing to unpack .../1-apparmor_2.11.0-3+deb9u2_amd64.deb ...
Unpacking apparmor (2.11.0-3+deb9u2) ...
Selecting previously unselected package snap-confine.
Preparing to unpack .../2-snap-confine_2.21-2+b1_amd64.deb ...
Unpacking snap-confine (2.21-2+b1) ...
Selecting previously unselected package ubuntu-core-launcher.
Preparing to unpack .../3-ubuntu-core-launcher_2.21-2+b1_amd64.deb ...
Unpacking ubuntu-core-launcher (2.21-2+b1) ...
Selecting previously unselected package squashfs-tools.
Preparing to unpack .../4-squashfs-tools_1%3a4.3-3+deb9u1_amd64.deb ...
Unpacking squashfs-tools (1:4.3-3+deb9u1) ...
Selecting previously unselected package snapd.
Preparing to unpack .../5-snapd_2.21-2+b1_amd64.deb ...
Unpacking snapd (2.21-2+b1) ...
Setting up squashfs-tools (1:4.3-3+deb9u1) ...
Processing triggers for systemd (232-25+deb9u6) ...
Processing triggers for man-db (2.7.6.1-2) ...
Setting up libapparmor-perl (2.11.0-3+deb9u2) ...
Setting up apparmor (2.11.0-3+deb9u2) ...
Created symlink /etc/systemd/system/sysinit.target.wants/apparmor.service → /lib/systemd/system/apparmor.service.
update-rc.d: warning: start and stop actions are no longer supported; falling back to defaults
diff: /var/lib/apparmor/profiles/.apparmor.md5sums: Tiedostoa tai hakemistoa ei ole
Setting up snap-confine (2.21-2+b1) ...
Setting up ubuntu-core-launcher (2.21-2+b1) ...
Setting up snapd (2.21-2+b1) ...
Created symlink /etc/systemd/system/multi-user.target.wants/snapd.autoimport.service → /lib/systemd/system/snapd.autoimport.service.
Created symlink /etc/systemd/system/timers.target.wants/snapd.refresh.timer → /lib/systemd/system/snapd.refresh.timer.
Created symlink /etc/systemd/system/sockets.target.wants/snapd.socket → /lib/systemd/system/snapd.socket.
Created symlink /etc/systemd/system/multi-user.target.wants/snapd.service → /lib/systemd/system/snapd.service.
Processing triggers for systemd (232-25+deb9u6) ...
$ sudo snap install gimp
error: cannot perform the following tasks:
- Mount snap "gimp" (94) (cannot find required base "core18")
$ snap run gimp
error: cannot find current revision for snap gimp: readlink /snap/gimp/current: no such file or directory
```

What's going on here? I feel like this is snap 101 stuff that I'm just not getting. Searching with the error message brought up nothing that seemed relevant.

**Comment by diddledan** [comment]
This is a bootstrap issue because you've only just installed snapd. You need to either install a different snap (one that doesn't use core18) first, or run `sudo snap install core`, to ensure that you have the latest core snap (not core18).

**Comment by terotil** [comment]
Now install succeeded. Gimp pukes a few warnings but other than that seems to be working nicely. Thanks!

```
$ snap run gimp
main.go:224: cannot change mount namespace of snap "gimp" according to change mount (/snap/gtk-common-themes/818/share/icons/Suru /snap/gimp/94/data-dir/icons/Suru none bind,ro 0 0): cannot use "/snap/gtk-common-themes/818/share/icons/Suru" as bind-mount source: not a directory
Gtk-Message: 21:42:01.661: Failed to load module "gail"
Gtk-Message: 21:42:01.661: Failed to load module "atk-bridge"
Gtk-Message: 21:42:01.707: Failed to load module "canberra-gtk-module"
/snap/gimp/94/usr/bin/gimp: Gimp-Widgets-WARNING: parse_iso_codes: error parsing '/build/gimp/parts/gimp/install/usr/share/xml/iso-codes/iso_639.xml': No such file or directory
```

That `cannot find required base "core18"` wasn't exactly helpful error message.

**Comment by diddledan** [comment]
OK, I'm going to close this issue then as the problem is not with the GIMP snap but snapd itself:

* https://forum.snapcraft.io/t/installation-fails-as-core-snap-not-found/9367
* should be fixed by snapcore/snapd#6118

The fix will make it's way to a future release of snapd.
